Add ZeroNet

zolagonano opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

**https://github.com/zolagonano/awesome-zeronet#readme**

ZeroNet is a distributed web-like peer-to-peer network, this list is going to help new users to find the sites, tools, implementations, plugins, and guides on ZeroNet

Reviewed PRs:
- #2275 
- #2325 

New reviewed PRs:
- #2316 
- #2302 
- #2370 

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

sindresorhus commented:
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

sindresorhus commented:
Also:

> Just commenting “looks good” or simply marking the pull request as approved does not count! You have to actually point out mistakes or improvement suggestions.

nyquist commented:
At the top of PR you describe the list, but you should describe ZeroNet

> Your entry here should include a short description about the project/theme of the list. It should not describe the list itself.

bytecauldron commented:
Hi @zolagonano,

Here are some of the formatting stuff I noticed with your list.

- Includes a project logo/illustration whenever possible. Please include a high-dpi image at the top of your list that links back to [ZeroNet](https://zeronet.io/).
- Your Awesome badge should be placed on the right side of the readme heading. Or, if you add the header image, it can also be centered if the list has a centered graphics header.
- The repo should have **awesome-list** & awesome as [GitHub topics](https://help.github.com/articles/about-topics).
- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none. I currently see third-level nested lists.
- Do not add the license name, text, or a Licence section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
- You need to also comment on your pull request with a secret word. Please read the pull request template again to find out what it is.

I noticed you are using markdown tables for your list entries. I understand why you formatted it this way but it might cause a ton of errors when using the linter.

Hope that helps. 👍

Lissy93 reviewed (changes requested):
You've also got the [following lint warnings](https://github.com/sindresorhus/awesome/runs/7378933323?check_suite_focus=true)

If it helps, you can run `npx awesome-lint` within your project to work through any errors locally.

zolagonano commented:
Thanks for the feedback @bytecauldron / @Lissy93 i will fix those soon :)

zolagonano commented:
> You've also got the [following lint warnings](https://github.com/sindresorhus/awesome/runs/7378933323?check_suite_focus=true)
> 
> If it helps, you can run `npx awesome-lint` within your project to work through any errors locally.

Fixed the lint errors, thanks

zolagonano commented:
@bytecauldron thanks for the help, fixed those issues that you mentioned :)

zolagonano commented:
There are some unmaintained sites in the list, but i have included a note for them.
They'll be moved to another markdown file or just get removed as the users switch to new versions of those sites,

zolagonano commented:
unicorn

0Lola commented:
@zolagonano 
I run `npx awesone-lint` to check [awesome-zeronet readme](https://github.com/zolagonano/awesome-zeronet), it already pass 🎉. 
You could comment `unicorn` to verify pull request again.

zolagonano commented:
@sindresorhus unicorn

HarshShukla01 commented:
Great Idea 💡

zrosenbauer reviewed (commented):
Status check still failing. 

I opened a PR on your repo to add in CI that you can use to verify it will pass (same thing this repo uses)

[https://github.com/zolagonano/awesome-zeronet/pull/2](https://github.com/zolagonano/awesome-zeronet/pull/2)

sindresorhus commented:
`Url` => `URL`

`Zeronet` => `ZeroNet`

`zeronet` => `ZeroNet`

---

The use of tables for everything makes the list very messy and hard to navigate.

zolagonano commented:
@sindresorhus Thanks, fixed those typos, but currently I don't have a replacement for tables

sindresorhus commented:
I also don't see the point of the "Active Maintenance" column. If it's not maintained, it's not awesome, and should not be included in the list. If you really want it included, it should be a separate Markdown file.

sindresorhus commented:
If you insist on keeping the tables (which I would strongly recommend moving away from), at least make the "description" come after the "name".

zolagonano commented:
> I also don't see the point of the "Active Maintenance" column. If it's not maintained, it's not awesome, and should not be included in the list. If you really want it included, it should be a separate Markdown file.

Thanks, removed "Active Maintenance" column and moved unmaintained projects to another markdown file, Also moved "Description" column next to name column.

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1624659433367429120
